src/Markup/Avalonia.Markup.Xaml/XamlIl/Runtime/XamlIlRuntimeHelpers.cs

```diff
@@ -51,6 +51,20 @@ public static class XamlIlRuntimeHelpers
             return new PointerDeferredContent<T>(resourceNodes, rootObject, parentScope, typedBuilder);
         }
 
+        // The builder is typed as IntPtr instead of delegate*<IServiceProvider, object> because Reflection.Emit has
+        // trouble with generic methods containing function pointers. See https://github.com/dotnet/runtime/issues/100020
+        public static unsafe INotSharedDeferredContent NotSharedDeferredTransformationFactoryV0<T>(
```

Comment:
You can delete this. As mentioned in the `ResourceDictionary` comments, we can reuse all the existing methods and types here and simply flag the non-shared resources inside the dictionary itself.